### PR TITLE
dub: add `url-ready` command

### PR DESCRIPTION
**Pull request by miguno**

### CODE REVIEW

File: debian/base/include/dub [hunk 1]
```diff
@@ -205,11 +252,15 @@ if __name__ == '__main__':
     check_env = actions.add_parser('ensure', description='Check if env var exists.')
     check_env.add_argument('name', help='Name of env var.')
 
-    check_env = actions.add_parser('wait', description='wait for network service to appear.')
+    check_env = actions.add_parser('wait', description='Wait for network service to appear.')
     check_env.add_argument('host', help='Host.')
-    check_env.add_argument('port', help='Host.', type=int)
+    check_env.add_argument('port', help='Port.', type=int)
```
Comment (miguno) [inline debian/base/include/dub]: Unrelated fix (wrong argument description)

File: debian/base/include/dub [hunk 2]
```diff
@@ -205,11 +252,15 @@ if __name__ == '__main__':
     check_env = actions.add_parser('ensure', description='Check if env var exists.')
     check_env.add_argument('name', help='Name of env var.')
 
-    check_env = actions.add_parser('wait', description='wait for network service to appear.')
+    check_env = actions.add_parser('wait', description='Wait for network service to appear.')
```
Comment (miguno) [inline debian/base/include/dub]: Unrelated fix: Consistent formatting.

File: debian/base/include/dub [hunk 3]
```diff
@@ -126,6 +128,51 @@ def wait_for_service(host, port, timeout):
...
+        port = 443
+
+    # Check if you can connect to the endpoint
+    status = wait_for_service(host, port, timeout)
+
+    if status:
+
+        # Check if service is responding as expected
+        r = requests.get(url)
+        if r.status_code // 100 == 2:
```
Comment (theduderog) [inline debian/base/include/dub]: is this any clearer than `r.status_code  == 200`?

File: debian/base/include/dub [hunk 4]
```diff
@@ -205,11 +252,15 @@ if __name__ == '__main__':
...
     check_env.add_argument('name', help='Name of env var.')
 
-    check_env = actions.add_parser('wait', description='wait for network service to appear.')
+    check_env = actions.add_parser('wait', description='Wait for network service to appear.')
     check_env.add_argument('host', help='Host.')
-    check_env.add_argument('port', help='Host.', type=int)
+    check_env.add_argument('port', help='Port.', type=int)
     check_env.add_argument('timeout', help='timeout in secs.', type=float)
 
+    check_env = actions.add_parser('url-ready', description='Wait for a URL to be retrievable.')
```
Comment (theduderog) [inline debian/base/include/dub]: should we call this `http-ready`?  That's what it is.  We're not handling any other types of URLs

File: debian/base/include/dub (hunk 3 above)
Comment (miguno) [inline debian/base/include/dub]: That part of the code is copied from `cub`'s `kr-ready`.  I actually considered to provide an optional parameter, `status_code` with a default of `200`, but eventually decided to keep things simple for the time being, and the idea to accept any `2xx` codes sounded like a good idea (which I assume was the reason for the related function in `cub`).

I'm happy to change this if need be.

File: debian/base/include/dub (hunk 4 above)
Comment (miguno) [inline debian/base/include/dub]: +1

File: debian/base/include/dub (hunk 3 above)
Comment (theduderog) [inline debian/base/include/dub]: oh, right.  makes sense as is.

**Comment by theduderog** [comment]
@miguno Feel free to merge.  The jenkins build is badly configured and always fails :(

**Comment by miguno** [comment]
Should I also merge this into `master`?  I have noticed that `3.2.x` hasn't been merged into `master` since 6 weeks, but I suppose it's fair game to just to a `git checkout master && git merge 3.2.x`, right?

**Comment by theduderog** [comment]
yes, 3.2.x should be merged into master  cc @aayars
